src/OrbitGl/App.cpp

```diff
@@ -940,17 +940,22 @@ void OrbitApp::Disassemble(uint32_t pid, const FunctionInfo& function) {
 }
 
 void OrbitApp::ShowSourceCode(const orbit_client_data::FunctionInfo& function) {
+  ScopedMetric metric{metrics_uploader_,
+                      orbit_metrics_uploader::OrbitLogEvent::ORBIT_SOURCE_CODE_SHOW};
+
   const ModuleData* module = module_manager_->GetModuleByPathAndBuildId(function.module_path(),
                                                                         function.module_build_id());
 
   auto loaded_module = RetrieveModuleWithDebugInfo(module);
 
   loaded_module
-      .ThenIfSuccess(
+      .Then(
           main_thread_executor_,
-          [this, module,
-           function](const std::filesystem::path& local_file_path) -> ErrorMessageOr<void> {
-            const auto elf_file = orbit_object_utils::CreateElfFile(local_file_path);
+          [this, module, function](const ErrorMessageOr<std::filesystem::path>& local_file_path)
+              -> ErrorMessageOr<void> {
+            if (local_file_path.has_error()) return local_file_path.error();
+
+            const auto elf_file = orbit_object_utils::CreateElfFile(local_file_path.value());
```

Comment:
I think the change of these lines is actually not necessary.. or is it?

Reply:
If we use `ThenIfSuccess`, then we miss this error case and not setting the metric status as internal error

Reply:
I don't think that is the case. In simplified this is:
```
loaded_module.ThenIfSuccess(A).Then(B)
```
if the `loaded_module` future is successful, then A is executed and afterwards B is executed. 
if the `loaded_module` future is an error, then A is _not_ executed, but B _is_ still executed.

If B would not get executed when `loaded_module` has an error, then the user would not see the error. If that is the case, then we have a bigger problem that needs to be fixed

Reply:
Thanks for the explanation! I changed here and below back to use `ThenIfSuccess`